Fixed issues related with Ansible and Python from RHSCL

fgiorgetti opened the pull request:
* Adjusted missing PIP modules
* Allow using a specific Python interpreter with Ansible
* Converter firewalld module tasks to firewalld-cmd command line (not working on RHEL8)

fgiorgetti committed:
Installing pip modules on router nodes that depends on docker

fgiorgetti committed:
Fixed issues with firewalld and python interpreter

* We were facing issues when using python 3 from scl

fgiorgetti commented:
@enkeys Any comments on this one? If not, I will merge it tomorrow.

nicob87 commented on integration/int_3Ri_2BhaRi2_3Re_2BhaRe3/brokers/tasks/copy_config_docker_host_edg.yml:
```diff
@@ -17,3 +17,6 @@
     dest: "{{ role_path }}/files/broker-edge/slave"
+
+- name: "Changing SELinux context to allow containers to mount volumes"
+  command: "chcon -Rt svirt_sandbox_file_t {{ role_path }}/files/broker-edge/"
```
maybe to avoid duplication 
```
tasks:
- import_tasks: tasks/change_selinux_ctx.yml
  vars:
    router_mode:edge/interior
```
and then using something like:
```
- name: "Changing SELinux context to allow containers to mount volumes"
  command: "chcon -Rt svirt_sandbox_file_t {{ role_path }}/files/broker-{{router_mode}}/"
```

fgiorgetti replied:
Maybe it wont be needed. Look at the PR proposed by @VeciCZ.
I think his suggestion is right and just mounting volumes with ":Z" might eliminate the need for changing the selinux context.

nicob87 commented on integration/int_3Ri_2BhaRi2_3Re_2BhaRe3/brokers/tasks/copy_config_docker_host_int.yml:
```diff
@@ -17,3 +17,6 @@
     dest: "{{ role_path }}/files/broker-interior/slave"
+
+- name: "Changing SELinux context to allow containers to mount volumes"
+  command: "chcon -Rt svirt_sandbox_file_t {{ role_path }}/files/broker-interior/"
```
the same that previous comment

nicob87 commented on integration/int_3Ri_2BhaRi2_3Re_2BhaRe3/deploy_ssh.yml:
```diff
@@ -68,6 +84,22 @@
+        name:
+          - docker
+          - selinux
+
```
also duplicated with the previous diff, this could be a task or a role, I do not know if it is possible to import tasks from within the same file...

fgiorgetti replied:
I have just added a new commit and removed the duplicated tasks.

nicob87 replied:
:ok:  :ok:

fgiorgetti committed:
Removed duplicated tasks based on PR review

Tasks common to docker_host, Router.I2 and Router.E3 have been unified:
- Common pip modules installation
- Linking python interpreter to /usr/bin
- Docker network creation

fgiorgetti committed:
Removed chcon calls as no longer needed

fgiorgetti committed:
Added tasks for installing required pip modules when using docker

fgiorgetti committed:
Replaced static pip binary with variable provided value

fgiorgetti committed:
Modified pip modules installation to be local to users

nicob87 commented on integration/int_3Ri_2BhaRi2_3Re_2BhaRe3/deploy_ssh.yml:
```diff
@@ -15,18 +15,38 @@
+        name:
+          - docker
+          - selinux
+        extra_args: --user
```
```
import_tasks: task/selinux_workaround.yml
```
or somethink like that? to use it here and up

fgiorgetti replied:
@nicob87 and @enkeys this has been applied and tested on both rhel7 and fedora 30.

fgiorgetti committed:
Created task files for common tasks used in both deployment plans

nicob87 reviewed (approved):
Thanks for the last commit, approved again.
